frmts/hdf5/hdf5drivercore.cpp

```diff
@@ -184,13 +184,23 @@ struct HDF5DriverSubdatasetInfo : public GDALSubdatasetInfo
 
             m_driverPrefixComponent = aosParts[0];
 
+            std::string part1{aosParts[1]};
+            if (part1.find('"') == 0)
```

Comment:
should make cppcheck happy
```suggestion
            if (!part1.empty() && part1[0] == '"')
```